src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs

```diff
@@ -171,7 +171,10 @@ public static void EnrichmentError(ILogger logger, Exception exception, string?
         var requestHeadersCount = record.RequestHeaders?.Count ?? 0;
         var responseHeadersCount = record.ResponseHeaders?.Count ?? 0;
 
-        var index = loggerMessageState.ReserveTagSpace(MinimalPropertyCount + statusCodePropertyCount + requestHeadersCount + responseHeadersCount);
+        var spaceToReserve = MinimalPropertyCount + statusCodePropertyCount + requestHeadersCount + responseHeadersCount +
+            record.PathParametersCount + (record.RequestBody is null ? 0 : 1) + (record.ResponseBody is null ? 0 : 1);
```

Comment:
Now that I've added support for the [TagName] property, can we get rid of this custom code and use the code generator instead?

Reply:
I have doubts regarding prefixes in header names, but I will try to convert this logic

Reply:
Done in #4842